client.go

```diff
@@ -119,11 +121,13 @@ func NewSnowthClient(addrs ...string) (*SnowthClient, error) {
 	// and manage the active/inactive lists accordingly
 	go sc.watchAndUpdate()
 
-	// for robustness, we will perform a discovery of associated nodes
-	// this works by pulling the topology information for given nodes
-	// and adding nodes discovered within the topology into the client
-	if err := sc.discoverNodes(); err != nil {
-		return nil, errors.Wrap(err, "failed to discover nodes")
+	if discover {
+		// for robustness, we will perform a discovery of associated nodes
+		// this works by pulling the topology information for given nodes
+		// and adding nodes discovered within the topology into the client
+		if err := sc.discoverNodes(); err != nil {
+			return nil, errors.Wrap(err, "failed to discover nodes")
+		}
```

Comment:
*opinion* it seems 'unfriendly' to me that a discovery failure will prevent the client for being created,